### ISSUE TITLE
Validate ID Uniqueness

### ISSUE THREAD

**Issue by kawazoe** [MobilityData/gtfs-validator#2060]
### Describe the problem

Our organisation consumes multiple GTFS feeds from different locations. Some of those feeds are somewhat hand-written. We have recently noticed that the validator does not produces an error if fields typed as "Unique ID", for which the specs says they must be unique within the file, are duplicated.

In our case, this happened because some of those IDs were human readable, and not just GUID or incremented indexes. Since the validator did not produce any errors, the publisher believed that IDs could repeat as long as the start/end dates did not overlap, which is not something that appears to be valid according to the specification.

### Describe the new validation rule

We propose to add a new validation rule that would ensure fields typed as "Unique ID" are in fact unique within the file and otherwise produce an error.

### Severity

ERROR

**Comment by skalexch** [comment]
@kawazoe can you please provide more detail on the exact file? (which GTFS file and which field).
If possible, can you please share the GTFS file?

**Comment by kawazoe** [comment]
The goal is to avoid scenarios like this one that we have seen recently in a calendar file:

<img width="719" height="202" alt="Image" src="https://github.com/user-attachments/assets/62332230-4964-484f-8e6e-603812de3ecf" />

Unfortunately, I only have this screenshot.

The specs states this about the ID field type (emphasis mine):

> ID - An ID field value is an internal ID, not intended to be shown to riders, and is a sequence of any UTF-8 characters. Using only printable ASCII characters is recommended. ***An ID is labeled "unique ID" when it must be unique within a file.*** IDs defined in one .txt file are often referenced in another .txt file. IDs that reference an ID in another table are labeled "foreign ID".

As a result, all of these fields would be impacted by this rule:

```
agency.txt - agency_id
stops.txt - stop_id
routes.txt - routes_id
trips.txt - trip_id
calendar.txt - service_id
fare_attributes.txt - fare_id
rider_categories.txt - rider_category_id
fare_media.txt - fate_media_id
areas.txt - area_id
networks.txt - network_id
pathways.txt - pathway_id
levels.txt - level_id
location_groups.txt - location_group_id
booking_rules.txt - booking_rule_id
attributions.txt - attribution_id
```

**Comment by skalexch** [comment]
@kawazoe you are correct about the spec. The validator itself [follows it](https://github.com/MobilityData/gtfs-validator/blob/435558c18b0f507e0cbb8c01f6f18c2ba9a2d3aa/main/src/main/java/org/mobilitydata/gtfsvalidator/table/GtfsCalendarSchema.java#L32) by specifying that `service_id` is the only field that constitutes the primary key of the file.

I tried replicating this by duplicating the Unique ID of a `calendar.txt` in a random GTFS from the Mobility Database and the [validator report](https://staging-gtfs-validator-results.mobilitydata.org/6f47508a-9374-46d4-ad1e-981f6a95fcc7/report.html) shows a duplicate key.

This might be a bug. If you are using a local validator version, try to use the [web version](https://gtfs-validator.mobilitydata.org/) and see if the notice does not come up. Because the issue you are mentioning should definitely be detected.

**Comment by kawazoe** [comment]
@skalexch I'll take a look at the web validator next time I have a chance, but if you say it's already supposed to catch it then I expect something else has happened on our end. Thanks!
